OpenRA.Mods.Common/Widgets/ProductionTabsWidget.cs

```diff
@@ -105,6 +105,8 @@ public class ProductionTabsWidget : Widget
 		readonly Lazy<ProductionPaletteWidget> paletteWidget;
 		string queueGroup;
 
+		readonly Dictionary<ProductionQueue, bool> cachedProductionQueueEnabledStates = new();
```

Comment:
Nit: Since you don't need any dictionary lookups, we could get away with `List<(ProductionQueue Queue, bool Enabled)>`